test/test-suite-1/test.mod.js

```diff
@@ -75,14 +78,15 @@ module.exports = function (dbType, context) {
             rank: {$mod: [0, 0]}
           },
           sort: ['name']
-        })
-        .catch(function (err) {
-          if (dbType === 'http') {
-            err.message.should.match(/Database encountered an unknown error/);
-            return;
-          }
+        }).then(function () {
+          throw new Error('expected an error here');
+        }, function (err) {
+            if (dbType === 'http') {
+              should.exist(err);
+              return;
+            }
 
-          err.message.should.match(/Bad divisor/);
+            err.message.should.match(/Bad divisor/);
```

Comment:
the previous version of this was not only relying on exact string names for error messages from the server, it was also incorrectly _not_ failing in the case where the function resolved successfully (i.e. with no Error). this fixes that